src/ThermalCameraSensor.cc

```diff
@@ -254,6 +262,29 @@ bool ThermalCameraSensor::CreateCamera()
       this->Name());
   this->dataPtr->thermalCamera->SetImageWidth(width);
   this->dataPtr->thermalCamera->SetImageHeight(height);
+  switch (pixelFormat)
+  {
+    case sdf::PixelFormatType::L_INT8:
+    {
+      this->dataPtr->thermalCamera->SetImageFormat(rendering::PF_L8);
+      // sanity check for resolution. The default resolution 10mK is too high
+      // for 8 bit cameras since it can only capture a temperature range of
+      // 0 to 2.55 degrees Kelvin ((2^8-1) * 0.01 = 2.55).
+      if (this->dataPtr->resolution < 1.0)
+      {
+        ignwarn << "8 bit thermal camera image format selected. "
+                << "The temperature linear resolution needs to be higher "
+                << "than 1.0. Defaulting to 3.0, range = [0, 255*3] K"
+                << std::endl;
+        this->dataPtr->resolution = 3.0;
+      }
+      break;
+    }
+    case sdf::PixelFormatType::L_INT16:
+    default:
+      this->dataPtr->thermalCamera->SetImageFormat(rendering::PF_L16);
+      break;
```

Comment:
```suggestion
    case sdf::PixelFormatType::L_INT16:
         this->dataPtr->thermalCamera->SetImageFormat(rendering::PF_L16);
         break;
   default:
         ignwarn << "Format not supported" << std::endl;
         return;
```

Reply:
this is actually on purpose. I default it to 16 bit if the format is not recognized. It also preserves existing behavior.